### PR TITLE
test(compiler-cli): convert components & directives compliance tests

### PR DESCRIPTION
**Pull request by JoostK**
This commit converts the components & directives compliance tests taken
from `r3_compiler_compliance_spec.ts` to the new test runner.

### CODE REVIEW

File: packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/value_composition/complex_selectors_other_directive_dir.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+OtherDirective.ɵdir = $r3$.ɵɵdefineDirective({
+  type: OtherDirective,
+  selectors: [["", 5, "span", "title", "", 9, "baz"]]
```
Comment (petebacondarwin) [inline packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/value_composition/complex_selectors_other_directive_dir.js:3]: We could be very snazzy and change these constants to use a `SelectorFlags` macro...

File: packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/content_projection/ng_content_with_structural_dir.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+SimpleComponent.ɵcmp = $r3$.ɵɵdefineComponent({
+  type: SimpleComponent,
+  selectors: [["simple"]],
+  ngContentSelectors: $c0$,
+  decls: 1,
+  vars: 1,
+  consts: [[4, "ngIf"]],
```
Comment (petebacondarwin) [inline packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/content_projection/ng_content_with_structural_dir.js:7]: `__AttributeMarker....__`?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks @JoostK. Another great PR. Can you add the missing attribute markers?

File: packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/value_composition/complex_selectors_other_directive_dir.js (hunk 1 above)
Comment (JoostK) [reply packages/compiler-cli/test/compliance/test_cases/r3_compiler_compliance/components_and_directives/value_composition/complex_selectors_other_directive_dir.js:3]: Well spotted! I noticed the same pattern for things like `ViewEncapsulation` and possible also `RenderFlags`. I'd consider this to be something to cleanup in follow ups.

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Caretaker - it looks like it is only this change that has triggered a presubmit: https://github.com/angular/angular/pull/39920/files?file-filters%5B%5D=.ts#diff-b844d7826ec99eed11ffc27fb0fcb1e8c0340fceea4cd616ed07e717f2997cbc

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:345508227:BASE:345508257:1607026768450:68117d3c).

**Pull request closed without merging**